### ISSUE TITLE
Impl Deref<&Value> for Sample

### ISSUE THREAD

**Issue by Mallets** [eclipse-zenoh/zenoh#241]
By implementing `Deref<&Value> for Sample` becomes possible to have an automatic shortcut to access the actual payload in samples. 

I.e., in addition to `sample.value.payload.contigous()` it will also possible to access the payload by typing `sample.payload.contigous()`.

**Comment by Mallets** [comment]
https://github.com/eclipse-zenoh/zenoh/pull/258 has been merged into api-changes branch.
This issue will be automatically closed once api-changes is merged into master.
